.gitlab/source_test/macos.yml

```diff
@@ -51,3 +51,69 @@ lint_macos:
     - !reference [.setup_python_mirror_linux]
     - python3 -m pip install -r tasks/libs/requirements-github.txt
     - inv -e github.trigger-macos --workflow-type "lint" --datadog-agent-ref "$CI_COMMIT_SHA" --python-runtimes "$PYTHON_RUNTIMES" --version-cache "$VERSION_CACHE_CONTENT"
+
+.macos_gitlab_amd64:
+  variables:
+    PYTHON_RUNTIMES: "3"
+  # The Gitlab macOS runners are currently long runners, so we need to clean them beforehand.
+  before_script:
+    # Remove the Go cache and env if the Go version changed
+    - |
+      GO_REPO_VERSION=$(cat .go-version)
+      GO_VERSION=$(go version | awk '{print $3}' | sed 's/go//')
+      if [ "$GO_REPO_VERSION" != "$GO_VERSION" ]; then
+        echo "Go version $GO_VERSION is different from $GO_REPO_VERSION in .go-version. Cleaning the environment."
+        go clean -cache -modcache -testcache
+        rm -rf $HOME/go/bin
+        echo "Installing Go $GO_REPO_VERSION..."
+        echo "$(gimme $(cat .go-version))"
+        eval $(gimme $(cat .go-version))
+      else
+        echo "Go current version $GO_VERSION is the same as .go-version. Keeping the cache."
+      fi
+    # Remove the Python cache and env if the Python version changed
+    - |
+      PYTHON_REPO_VERSION=$(cat .python-version)
+      PYTHON_VERSION=$(python3 --version | awk '{print $2}' | sed 's/\.[0-9]*$//')
+      if [ "$PYTHON_REPO_VERSION" != "$PYTHON_VERSION" ]; then
+        echo "Python version $PYTHON_VERSION is different from $PYTHON_REPO_VERSION in .python-version. Cleaning the environment."
+        pyenv uninstall -f datadog-agent
+        echo "Installing Python $PYTHON_REPO_VERSION..."
+        pyenv virtualenv 3.11.8 datadog-agent
+        pyenv activate datadog-agent
+      else
+        echo "Python current version $PYTHON_VERSION is the same as .python-version. Keeping the existing environment."
+      fi
+    # Installing the job dependencies
+    - python3 -m pip install -r requirements.txt -r tasks/libs/requirements-github.txt
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Don't we need `tasks/requirements.txt` instead of github requirements ?

Reply:
No idea that's how it was done in `datadog-agent-macos-build`. Checking